test/components/syncTest.js

```diff
@@ -201,3 +172,151 @@ describe('Sync Panel', function () {
+
+  function * setupSync (client, seed) {
+    yield setupBrave(Brave.app.client)
+    yield Brave.app.client.saveSyncInitData(seed, Immutable.fromJS([0]), 0, 'data:image/png;base64,foo')
+
+    yield Brave.app.client
+      .windowByUrl(Brave.browserWindowUrl)
+      .tabByIndex(0)
+      .loadUrl(prefsUrl)
+      .waitForVisible(syncTab)
+      .click(syncTab)
+      .waitForVisible(syncSwitch)
+      .click(syncSwitch)
+      .windowByUrl(Brave.browserWindowUrl)
+      .waitUntil(function () {
+        return this.getAppState().then((val) => {
+          return val.value.settings['sync.enabled'] === true
+        })
+      })
+      .pause(1000) // XXX: Wait for sync init
+  }
+
+  describe('bookmarks', function () {
+    Brave.beforeAllServerSetup(this)
+
+    function * bookmarkUrl (url, title) {
+      yield Brave.app.client
+        .tabByIndex(0)
+        .loadUrl(url)
+        .windowParentByUrl(url)
+        .activateURLMode()
+        .waitForVisible(navigatorNotBookmarked)
+        .click(navigatorNotBookmarked)
+        .waitForVisible(doneButton)
+        .setInputText('#bookmarkName input', title)
+        .waitForBookmarkDetail(url, title)
+        .waitForEnabled(doneButton)
+        .click(doneButton)
+    }
+
+    before(function * () {
+      this.page1Url = Brave.server.url('page1.html')
+      this.page1Title = 'Page 1'
+      this.page2Url = Brave.server.url('page2.html')
+      this.page2Title = 'Page 2'
+      this.page2TitleUpdated = 'All shall hail pyramids'
+      this.page3Url = Brave.server.url('page3.html')
+      this.page3Title = 'Page 3'
+      this.seed = new Immutable.List(crypto.randomBytes(32))
+      yield Brave.startApp()
+      yield setupSync(Brave.app.client, this.seed)
+
+      // For Create: Bookmark page 1
+      yield bookmarkUrl(this.page1Url, this.page1Title)
+
+      // For Update: Bookmark page 2 and rename it
+      yield Brave.app.client
+        .tabByIndex(0)
+        .waitForUrl(this.page1Url)
+      yield bookmarkUrl(this.page2Url, this.page2Title)
+      yield Brave.app.client
+        .pause(500) // XXX: Wait for sync
+        .activateURLMode()
+        .waitForVisible(navigatorBookmarked)
+        .click(navigatorBookmarked)
+        .waitForVisible(doneButton)
+        .waitForExist('#bookmarkName input')
+        .setInputText('#bookmarkName input', this.page2TitleUpdated)
+        .waitForBookmarkDetail(this.page2Url, this.page2TitleUpdated)
+        .waitForEnabled(doneButton)
+        .click(doneButton)
+
+      // For Delete: Bookmark page 3 and delete it
+      yield Brave.app.client
+        .tabByIndex(0)
+        .waitForUrl(this.page2Url)
+      yield bookmarkUrl(this.page3Url, this.page3Title)
+      yield Brave.app.client
+        .pause(500) // XXX: Wait for sync
+        .activateURLMode()
+        .waitForVisible(navigatorBookmarked)
+        .click(navigatorBookmarked)
+        .waitForExist(removeButton)
+        .click(removeButton)
+
+      // XXX: Wait for sync
```

Comment:
it would be helpful for these comments to mention what part of sync they are waiting for